cpp/src/arrow/type-test.cc

```diff
@@ -399,6 +399,40 @@ TEST(TestStructType, Basics) {
   // TODO(wesm): out of bounds for field(...)
 }
 
+TEST(TestStructType, GetChildByName) {
+  auto f0 = field("f0", int32());
+  auto f1 = field("f1", uint8(), false);
+  auto f2 = field("f2", utf8());
+  auto f3 = field("f3", list(int16()));
+
+  StructType struct_type({f0, f1, f2, f3});
+  std::shared_ptr<Field> result;
+
+  result = struct_type.GetChildByName("f1");
+  ASSERT_TRUE(f1->Equals(result));
```

Comment:
You could actually check for pointer equality here, i.e. `ASSERT_EQ(f1, result)`.